README.md

```diff
@@ -85,872 +118,382 @@ Unlike many authorization systems where a service controls access to resources i
+A resource describes the noun of a capability. It is either The resource pointer MUST be provided in [URI] format. Arbitrary and custom URIs MAY be used, provided that the intended recipient can decode the URI. The URI is merely a unique identifier to describe the pointer to — and within — a resource.
 
-#### Examples
+Having a unique agent represent a resource (and act as its manager) is RECOMMENDED. However, to help traditional ACL-based systems transition to certificate capabilities, an agent MAY manage multiple resources, and act as the registrant in the ACL system. These situations can be represented as follows:
 
-```json
-"nbf": 1529496683,
-"exp": 1575606941,
-```
+Unless explicitly stated, the Resource of a UCAN MUST be the Subject.
 
-### 3.2.4 Nonce
+## 3.4 Command
 
-The OPTIONAL nonce parameter `nnc` MAY be any value. A randomly generated string is RECOMMENDED to provide a unique UCAN, though it MAY also be a monotonically increasing count of the number of links in the hash chain. This field helps prevent replay attacks and ensures a unique CID per delegation. The `iss`, `aud`, and `exp` fields together will often ensure that UCANs are unique, but adding the nonce ensures this uniqueness.
+Commands are concrete messages ("verbs") that MUST be unambiguously interpretable by the Subject of a UCAN. Commands are REQUIRED in invocations. Some examples include `msg/send`, `crud/read`, and `ucan/revoke`.
 
-This field SHOULD NOT be used to sign arbitrary data, such as signature challenges. See the `fct` field for more.
+Much like other message-passing systems, the specific resource MUST define the behavior for a particular message. For instance, `crud/update` MAY be used to destructively update a database row, or append to a append-only log. Specific messages MAY be created at will; the only restriction is that the Executor understand how to interpret that message in the context of a specific resource.
 
-#### Examples
+While arbitrary semantics MAY be described, they MUST apply to the target resource. For instance, it does not make sense to apply `msg/send` to a typical file system. 
 
-``` json
-"nnc": "1701-D"
-```
+Commands MUST NOT be case-sensitive. There MUST be at least one path segment as a namespace. For example, `http/put` and `db/put` MUST be treated as unique from each other.
 
-### 3.2.5 Facts
+### 3.4.1 Ability
 
-The OPTIONAL `fct` field contains a map of arbitrary facts and proofs of knowledge. The enclosed data MUST be self-evident and externally verifiable. It MAY include information such as hash preimages, server challenges, a Merkle proof, dictionary data, etc.
+Abilities abstract over [Command]s to allow for extension of UCAN delegations.
```

Comment:
Yeah this definitely needs to be tightened up. The difference is that a Command is something that can be called in an Invocation. You can't really call arguments on `http/*`, and adding any args to `http/*` make it much more difficult to delegate portion (`http/* -> http/put`) because it's really difficult to figure out how one would check the delegation.

I've been thinking about maybe just dropping the distinction and saying "too bad, you need to trace the `http/put` all the way to the root, you can't use `http/*` to prove it" but that seems like such a shame.